datafusion/expr/src/expr_schema.rs

```diff
@@ -37,7 +37,7 @@ use std::sync::Arc;
 /// trait to allow expr to typable with respect to a schema
 pub trait ExprSchemable {
     /// given a schema, return the type of the expr
-    fn get_type<S: ExprSchema>(&self, schema: &S) -> Result<DataType>;
+    fn get_type<S: ExprSchema>(&self, schema: &S) -> Result<LogicalType>;
```

Comment:
Replace all arrow `DataType` used in the logical planning stage with `LogicalType`(literal ScalarValue TBD).